src/components/SubNav/SubNav.tsx

```diff
@@ -0,0 +1,400 @@
+import React, { forwardRef, useEffect, useState, useRef } from "react";
+
+import {
+  Box,
+  chakra,
+  ChakraComponent,
+  useMultiStyleConfig,
+  Flex,
+} from "@chakra-ui/react";
+import Button from "../Button/Button";
+import Link from "../Link/Link";
+import List from "../List/List";
+import useNYPLBreakpoints from "../../hooks/useNYPLBreakpoints";
+
+export const actionBackgroundColorsArray = [
+  "brand.primary-05",
+  "section.blogs.primary-05",
+  "section.books-and-more.primary-05",
+  "section.connect.primary-05",
+  "section.education.primary-05",
+  "section.locations.primary-05",
+  "section.research.primary-05",
+  "section.research-library.lpa-05",
+  "section.research-library.schomburg-05",
+  "section.research-library.schwarzman-05",
+  "section.whats-on.primary-05",
+  "dark.brand.primary-05",
+  "dark.section.blogs.primary-05",
+  "dark.section.books-and-more.primary-05",
+  "dark.section.connect.primary-05",
+  "dark.section.education.primary-05",
+  "dark.section.locations.primary-05",
+  "dark.section.research.secondary-05",
+  "dark.section.research-library.lpa-05",
+  "dark.section.research-library.schomburg-05",
+  "dark.section.research-library.schwarzman-05",
+  "dark.section.whats-on.primary-05",
+] as const;
+
+export type actionBackgroundColors = typeof actionBackgroundColorsArray[number];
+
+export const highlightColorsArray = [
+  "brand.primary",
+  "section.blogs.primary",
+  "section.books-and-more.primary",
+  "section.connect.primary",
+  "section.education.primary",
+  "section.locations.primary",
+  "section.research.primary",
+  "section.research-library-lpa.primary",
+  "section.research-library-schomburg.primary",
+  "section.research-library-schwarzman.primary",
+  "section.whats-on.primary",
+  "dark.brand.primary",
+  "dark.section.blogs.primary",
+  "dark.section.books-and-more.primary",
+  "dark.section.connect.primary",
+  "dark.section.education.primary",
+  "dark.section.locations.primary",
+  "dark.section.research.secondary",
+  "dark.section.research-library-lpa.primary",
+  "dark.section.research-library-schomburg.primary",
+  "dark.section.research-library-schwarzman.primary",
+  "dark.section.whats-on.primary",
+];
+export type highlightColors = typeof highlightColorsArray[number];
+
+export interface SubNavProps {
+  /** Additional class name for the `SubNav` component. */
+  className?: string;
+  /**
+   * Primary actions displayed on the left side of the SubNav.
+   * Use SubNavButton and SubNavLink components, which mirror
+   * the DS Button and Link functionalities, plus the isOutlined prop.
+   */
+  primaryActions: ({
+    highlightColor,
+    actionBackgroundColor,
+    selectedItem,
+  }: {
+    highlightColor?: string;
+    actionBackgroundColor?: string;
+    selectedItem?: string;
+  }) => React.ReactNode;
```

Comment:
Why are these props passed into `primaryActions` and `secondaryActions`?

Reply:
It should be `(SubNavButton | SubNavLink)[]` but we want to move away from mapping and cloning children to reduce creating new components on the client. This render prop pattern allows the `SubNavButton` or `SubNavLink` children to be styled appropriately based on `SubNav`